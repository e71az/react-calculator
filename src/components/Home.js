import React from 'react';

const Home = () => (
  <div className="home-container ml-5 mt-5">
    <h1>Welcome to my page!</h1>
    <p>
      In this page you can navigate around and try out my self-made react
      calculator as well as checking a quote!
    </p>
    <p>
      Grab a cup of coffee, a donut and enjoy this awesome page that took some
      seconds to be created!
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla pretium
      ipsum mi. Vestibulum finibus urna at nibh accumsan varius. Vestibulum
      aliquam ornare libero non aliquam. Morbi gravida suscipit arcu vel dictum.
      Cras tincidunt et nisl sit amet rutrum. In cursus consequat diam, at
      hendrerit tortor elementum id. Maecenas tempus ipsum vel ligula tristique,
      tincidunt accumsan quam imperdiet. Nunc quis mauris maximus, elementum
      risus ut, accumsan sem. Cras euismod, est ac posuere porta, augue urna
      tincidunt justo, at dapibus sapien nulla nec mauris. Donec volutpat metus
      nibh, ornare commodo dui ornare nec.
    </p>

    <p>
      Sed vulputate elit purus, vel rutrum ex ullamcorper vitae. Vestibulum
      vehicula bibendum sapien, quis maximus purus tempor in. Nullam sed
      convallis erat. Curabitur vitae justo in nulla tempor egestas eu id nibh.
      Nulla ut mattis tortor, eu venenatis enim. In ac dui et nunc tincidunt
      placerat quis eu sapien. Duis euismod, dolor ac aliquam rutrum, purus
      sapien luctus ligula, eu convallis arcu eros sed ante. Integer congue
      facilisis quam, ac venenatis mi varius non. Mauris tristique sodales
      ornare. Proin pharetra volutpat ex ac venenatis. Cras nec odio feugiat,
      fermentum est ac, tristique purus. Suspendisse ac justo feugiat ligula
      fringilla fringilla. Mauris vel purus blandit, auctor felis vitae,
      eleifend lectus. Vestibulum id nibh sem. Phasellus facilisis ligula
      lectus, vitae consectetur massa suscipit eu.
    </p>
  </div>
);

export default Home;
