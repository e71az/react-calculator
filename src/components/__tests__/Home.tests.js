import React from 'react';
import renderer from 'react-test-renderer';
import MainNavbar from '../NavBar';
import Home from '../Home';

describe('Renders Home component', () => {
  it('Should render the navbar and home components correctly ', () => {
    const tree = renderer
      .create(
        <>
          <MainNavbar />
          <Home />
        </>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
