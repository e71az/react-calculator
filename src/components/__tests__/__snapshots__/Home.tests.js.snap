// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Home component Should render the navbar and home components correctly  1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#home"
    >
      Math Magicians
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="ml-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="/"
          href="/"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Home
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/calc"
          href="/calc"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Calculator
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/quote"
          href="/quote"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Quote
        </a>
      </div>
    </div>
  </nav>,
  <div
    className="home-container ml-5 mt-5"
  >
    <h1>
      Welcome to my page!
    </h1>
    <p>
      In this page you can navigate around and try out my self-made react calculator as well as checking a quote!
    </p>
    <p>
      Grab a cup of coffee, a donut and enjoy this awesome page that took some seconds to be created!
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla pretium ipsum mi. Vestibulum finibus urna at nibh accumsan varius. Vestibulum aliquam ornare libero non aliquam. Morbi gravida suscipit arcu vel dictum. Cras tincidunt et nisl sit amet rutrum. In cursus consequat diam, at hendrerit tortor elementum id. Maecenas tempus ipsum vel ligula tristique, tincidunt accumsan quam imperdiet. Nunc quis mauris maximus, elementum risus ut, accumsan sem. Cras euismod, est ac posuere porta, augue urna tincidunt justo, at dapibus sapien nulla nec mauris. Donec volutpat metus nibh, ornare commodo dui ornare nec.
    </p>
    <p>
      Sed vulputate elit purus, vel rutrum ex ullamcorper vitae. Vestibulum vehicula bibendum sapien, quis maximus purus tempor in. Nullam sed convallis erat. Curabitur vitae justo in nulla tempor egestas eu id nibh. Nulla ut mattis tortor, eu venenatis enim. In ac dui et nunc tincidunt placerat quis eu sapien. Duis euismod, dolor ac aliquam rutrum, purus sapien luctus ligula, eu convallis arcu eros sed ante. Integer congue facilisis quam, ac venenatis mi varius non. Mauris tristique sodales ornare. Proin pharetra volutpat ex ac venenatis. Cras nec odio feugiat, fermentum est ac, tristique purus. Suspendisse ac justo feugiat ligula fringilla fringilla. Mauris vel purus blandit, auctor felis vitae, eleifend lectus. Vestibulum id nibh sem. Phasellus facilisis ligula lectus, vitae consectetur massa suscipit eu.
    </p>
  </div>,
]
`;

exports[`Renders Home component Should render the navbar and quote components correctly  1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#home"
    >
      Math Magicians
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="ml-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="/"
          href="/"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Home
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/calc"
          href="/calc"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Calculator
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/quote"
          href="/quote"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Quote
        </a>
      </div>
    </div>
  </nav>,
  <div
    className="home-container ml-5 mt-5"
  >
    <h1>
      Welcome to my page!
    </h1>
    <p>
      In this page you can navigate around and try out my self-made react calculator as well as checking a quote!
    </p>
    <p>
      Grab a cup of coffee, a donut and enjoy this awesome page that took some seconds to be created!
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla pretium ipsum mi. Vestibulum finibus urna at nibh accumsan varius. Vestibulum aliquam ornare libero non aliquam. Morbi gravida suscipit arcu vel dictum. Cras tincidunt et nisl sit amet rutrum. In cursus consequat diam, at hendrerit tortor elementum id. Maecenas tempus ipsum vel ligula tristique, tincidunt accumsan quam imperdiet. Nunc quis mauris maximus, elementum risus ut, accumsan sem. Cras euismod, est ac posuere porta, augue urna tincidunt justo, at dapibus sapien nulla nec mauris. Donec volutpat metus nibh, ornare commodo dui ornare nec.
    </p>
    <p>
      Sed vulputate elit purus, vel rutrum ex ullamcorper vitae. Vestibulum vehicula bibendum sapien, quis maximus purus tempor in. Nullam sed convallis erat. Curabitur vitae justo in nulla tempor egestas eu id nibh. Nulla ut mattis tortor, eu venenatis enim. In ac dui et nunc tincidunt placerat quis eu sapien. Duis euismod, dolor ac aliquam rutrum, purus sapien luctus ligula, eu convallis arcu eros sed ante. Integer congue facilisis quam, ac venenatis mi varius non. Mauris tristique sodales ornare. Proin pharetra volutpat ex ac venenatis. Cras nec odio feugiat, fermentum est ac, tristique purus. Suspendisse ac justo feugiat ligula fringilla fringilla. Mauris vel purus blandit, auctor felis vitae, eleifend lectus. Vestibulum id nibh sem. Phasellus facilisis ligula lectus, vitae consectetur massa suscipit eu.
    </p>
  </div>,
]
`;

exports[`Renders Quote component Should render the navbar and quote components correctly  1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="#home"
    >
      Math Magicians
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="ml-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="/"
          href="/"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Home
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/calc"
          href="/calc"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Calculator
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/quote"
          href="/quote"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Quote
        </a>
      </div>
    </div>
  </nav>,
  <div
    className="home-container ml-5 mt-5"
  >
    <h1>
      Welcome to my page!
    </h1>
    <p>
      In this page you can navigate around and try out my self-made react calculator as well as checking a quote!
    </p>
    <p>
      Grab a cup of coffee, a donut and enjoy this awesome page that took some seconds to be created!
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla pretium ipsum mi. Vestibulum finibus urna at nibh accumsan varius. Vestibulum aliquam ornare libero non aliquam. Morbi gravida suscipit arcu vel dictum. Cras tincidunt et nisl sit amet rutrum. In cursus consequat diam, at hendrerit tortor elementum id. Maecenas tempus ipsum vel ligula tristique, tincidunt accumsan quam imperdiet. Nunc quis mauris maximus, elementum risus ut, accumsan sem. Cras euismod, est ac posuere porta, augue urna tincidunt justo, at dapibus sapien nulla nec mauris. Donec volutpat metus nibh, ornare commodo dui ornare nec.
    </p>
    <p>
      Sed vulputate elit purus, vel rutrum ex ullamcorper vitae. Vestibulum vehicula bibendum sapien, quis maximus purus tempor in. Nullam sed convallis erat. Curabitur vitae justo in nulla tempor egestas eu id nibh. Nulla ut mattis tortor, eu venenatis enim. In ac dui et nunc tincidunt placerat quis eu sapien. Duis euismod, dolor ac aliquam rutrum, purus sapien luctus ligula, eu convallis arcu eros sed ante. Integer congue facilisis quam, ac venenatis mi varius non. Mauris tristique sodales ornare. Proin pharetra volutpat ex ac venenatis. Cras nec odio feugiat, fermentum est ac, tristique purus. Suspendisse ac justo feugiat ligula fringilla fringilla. Mauris vel purus blandit, auctor felis vitae, eleifend lectus. Vestibulum id nibh sem. Phasellus facilisis ligula lectus, vitae consectetur massa suscipit eu.
    </p>
  </div>,
]
`;
